src/models/generic/WorkflowModel.ts

```diff
@@ -1029,4 +1052,176 @@ export abstract class WorkflowModel extends ValidationBase implements Serializab
+        const sourceType = source.type;
+
+        // Check file types, true if at least one source file type is in destination file types
+        const checkFileTypes = () => {
+            if (source.fileTypes.length && !source.fileTypes
+                    .some(sourceType => !!destination.fileTypes
+                        .find((destType) => sourceType.toLowerCase() === destType.toLowerCase()))) {
+                return false;
+            } else {
+                return true;
+            }
+        };
+
+        // If type is the same (string -> string...) or type is array and items has the same type ([string] -> [string]...)
+        if (destinationType.type === sourceType.type && destinationType.items === sourceType.items) {
+
+            if (destinationType.type === "File" || destinationType.items === "File") {
+                isValid = checkFileTypes();
+            }
+
+        } else {
+
+            // If destination is array of items of type source (string -> [string]...)
+            if (destinationType.items === sourceType.type) {
+                isValid = checkFileTypes();
+            } else {
+                isValid = false;
+            }
+        }
+
+        // Whether to update connections or not (false when called from connectSource function)
+        if (updateConnection) {
+            this.connections.filter((c) =>
+                    c.isVisible && (c.destination.id === destination.connectionId && c.source.id === source.connectionId))
+                .forEach((c) => {
+                c.isValid = isValid;
+            });
+        }
+
+        if (!isValid) {
+            // Set warning
+            destination.updateValidity({[destination.loc + ".sources[" + source.sourceId + "]"]: {
+                message: `invalid connection (type or file types mismatch)`,
+                type: "warning"
+            }});
+        }
+
+        return isValid;
+    }
+
+    /**
+     * Validate all connections made with given destination
+     */
+    private validateDestination(destination: WorkflowOutputParameterModel | WorkflowStepInputModel, updateConnection = false) {
+
+        destination.cleanValidity();
+
+        // Find all sources connected to given destination
+        const sources = this.connections.filter((connection) => {
+            return connection.isVisible && (connection.destination.id === destination.connectionId);
+        }).map((connection) => {
+            return this.findById(connection.source.id);
+        });
+
+        // Validate all connections
+        sources.forEach((source) => {
+           this.validateConnection(destination, source, updateConnection);
+        });
+
+    }
+
+    /**
+     * Validate all connections made with given IO port
+     */
+    validateConnectionsForIOPort(port: WorkflowOutputParameterModel | WorkflowInputParameterModel) {
+
+        const isInput = port instanceof WorkflowInputParameterModel;
+
+        if (!isInput) {
```

Comment:
Unnecessary declaration, `if (port instanceof WorkflowInputParameterModel)` is fine.